CONTRIBUTING.md

```diff
@@ -2,33 +2,79 @@
+For each of our repositories we use the same model for contributing code. Developers wanting to  contribute must create pull requests. This process is described in the GitHub [Creating a pull request from a fork](https://help.github.com/en/articles/creating-a-pull-request-from-a-fork) documentation. Each pull request should be initiated against the master branch in the Algorand repository.  After a pull request is submitted the core development team will review the submission and communicate with the developer using the comments sections of the PR. After the submission is reviewed and approved, it will be merged into the master branch of the source. These changes will be merged to our release branch on the next viable release date.
 
-# Code Guidelines
+## Code Guidelines
 
 We make a best-effort attempt to adhere to [PEP 8 - Style Guide for Python Code](https://www.python.org/dev/peps/pep-0008/).  Keep the following context in mind:
 * Our default stance is to run linter checks during the build process.
 * Notable exception:  [naming conventions](https://peps.python.org/pep-0008/#naming-conventions).
 
-## Naming Convention Guidelines
+### Naming Convention Guidelines
 Since PyTeal aims for backwards compatibility, it's _not_ straightforward to change naming conventions in public APIs.  Consequently, the repo contains some deviations from PEP 8 naming conventions.
 
 In order to retain a consistent style, we prefer to continue deviating from PEP 8 naming conventions in the following cases.  We try to balance minimizing exceptions against providing a consistent style for existing software.
 * Enums - Define with lowercase camelcase.  Example:  https://github.com/algorand/pyteal/blob/7c953f600113abcb9a31df68165b61a2c897f591/pyteal/ast/txn.py#L37
 * Factory methods - Define following [class name](https://peps.python.org/pep-0008/#class-names) conventions.  Example:  https://github.com/algorand/pyteal/blob/7c953f600113abcb9a31df68165b61a2c897f591/pyteal/ast/ternaryexpr.py#L63
 
 Since it's challenging to enforce these exceptions with a linter, we rely on PR creators and reviewers to make a best-effort attempt to enforce agreed upon naming conventions.
+
+### Module Guidelines
+
+Every directory containing source code should be a Python module, meaning it should have an `__init__.py` file. This `__init__.py` file is responsible for exporting all public objects (i.e. classes, functions, and constants) for use outside of that module.
+
+Modules may be created inside of other modules, in which case the deeper module is called a submodule or child module, and the module that contains it is called the parent module. For example, `pyteal` is the parent module to `pyteal.ast`.
+
+A sibling module is defined as a different child module of the parent module. For example, `pyteal.ast` and `pyteal.ir` are sibling modules.
+
+### Import Guidelines
```

Comment:
Leaving a note to document a verbal with @jasonpaulos:

* There _might_ be a flake8 plugin we can use to mechanically check the import guidelines:  https://pypi.org/project/flake8-absolute-import/. 
* I've identified an option exists, but didn't evaluate it.  @jasonpaulos will take a quick look.  If it works out-of-the-box and looks straight-forward, then we can include it here.  Otherwise, we don't want to hold the PR up.

Reply:
Sorry - I confused myself.  There's 2 options that might work for us.  Same disclaimer as above:
* https://pypi.org/project/flake8-tidy-imports/
* https://pypi.org/project/flake8-absolute-import/

Both projects appear to be small scale though recently updated.  A quick scan suggests _flake8-tidy-imports_ is a bit more flexible.  Unclear if that's desirable.

Reply:
Added `flake8-tidy-imports`, thanks for the suggestion